packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemFileAdapter.kt

```diff
@@ -0,0 +1,101 @@
+package expo.modules.filesystem.next
+
+import android.content.Context
+import android.net.Uri
+import androidx.core.net.toUri
+import androidx.documentfile.provider.DocumentFile
+import java.io.File
+import java.net.URI
+
+class FileSystemFileAdapter(val context: Context, val uri: Uri) {
+  val javaFile get() = File(URI.create(uri.toString()))
```

Comment:
```suggestion
  val javaFile = File(URI.create(uri.toString()))
```
Not sure if creating a new file each time you call this property is a good idea.

Reply:
I've tried making these accesses smarter in the latest commit.